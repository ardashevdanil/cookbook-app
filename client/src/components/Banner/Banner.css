.Banner {
  height: 471px;
  margin: 81px 0;
  box-shadow: 0 0 13px 0 rgba(11, 12, 12, 0.15);
  background: no-repeat center right;
  background-size: contain;
  position: relative;
  z-index: 1;
}

.Banner__content {
  min-height: calc(100% - 104px);
  width: 370px;
  padding: 52px 0 52px 94px;
  background-color: #f8f8f8;
}

.Banner__logo {
  height: 105px;
  background: url(../../logo.png) no-repeat;
  background-size: contain;
}

.Banner__name {
  font-family: 'Playfair Display', serif;
  font-size: 25px;
  color: #911026;
  margin: 25px -2px;
}

.Banner__description {
  width: 320px;
  font-size: 15px;
  margin-bottom: 27px;
}

.Banner__button {
  height: 27px;
  width: 103px;
  border: solid 1px #c2c2c2;
  font-size: 12px;
  text-align: center;
  text-decoration: none;
  color: inherit;
  display: table-cell;
  vertical-align: middle;
  cursor: pointer;
  user-select: none;
}

.Banner__button:hover {
  color: #911026;
  border-color: #911026;
}

.Banner__change-menu {
  width: 76px;
  height: 10px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  position: absolute;
  bottom: 20px;
  right: 18px;
}

.Banner__nav-button {
  width: 11.8%;
  height: 90%;
  border-radius: 100%;
  background-color: white;
  cursor: pointer;
}

.Banner__nav-button:hover, .Banner__nav-button_active {
  background-color: #555;
}

@media only screen and (max-width: 1120px) {
  .Banner {
    height: 63.4vw;
    margin: 0 0 21.6vw 0;
    box-shadow: none;
    background: no-repeat center center;
    background-size: cover;
  }

  .Banner__logo,
  .Banner__name,
  .Banner__description {
    display: none;
  }

  .Banner__content {
    width: auto;
    padding: 0;
    background: transparent;
    position: relative;
    top: 63.4vw;
  }
  
  .Banner__button {
    width: 100%;
    height: 21.6vw;
    border: none;
    background-color: #f8f8f8;
    font-size: 5.5vw;
    display: block;
    box-sizing: border-box;
    padding-top: 7.5vw;
  }

  .Banner__change-menu {
    width: 23.8vw;
    height: 3.1vw;
  }
}