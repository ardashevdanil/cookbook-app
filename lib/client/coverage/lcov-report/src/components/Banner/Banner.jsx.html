<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Banner/Banner.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/Banner</a> Banner.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">76% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>19/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">62.5% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>18/24</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';
import './Banner.css';
&nbsp;
class Banner extends React.Component {
  constructor(props) {
    super(props);
    this.changeBanner = this.changeBanner.bind(this);
    this.state = {
      selectedBanner: 0,
    };
    this.handleChangeBanner = this.handleChangeBanner.bind(this);
    this.timer = setInterval(this.changeBanner, 5000);
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  co</span>mponentWillUnmount() {
<span class="cstat-no" title="statement not covered" >    clearInterval(this.timer);</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  ch</span>angeBanner() {
<span class="cstat-no" title="statement not covered" >    this.setState(<span class="fstat-no" title="function not covered" >(p</span>revState) =&gt; {</span>
      let nextBanner;
&nbsp;
<span class="cstat-no" title="statement not covered" >      if (prevState.selectedBanner === 4) {</span>
<span class="cstat-no" title="statement not covered" >        nextBanner = 0;</span>
      } else {
<span class="cstat-no" title="statement not covered" >        nextBanner = { ...prevState }.selectedBanner + 1;</span>
      }
&nbsp;
<span class="cstat-no" title="statement not covered" >      return { selectedBanner: nextBanner };</span>
    });
  }
&nbsp;
  handleChangeBanner(id) {
    clearInterval(this.timer);
    this.setState({ selectedBanner: id });
  }
&nbsp;
  render() {
    const { selectedBanner } = this.state;
    const { items } = this.props;
&nbsp;
    if (!items.length) return null;
&nbsp;
    const navButtons = [0, 1, 2, 3, 4].map((item) =&gt; {
      let className;
&nbsp;
      if (selectedBanner === item) {
        className = 'Banner__nav-button_active Banner__nav-button';
      } else {
        className = 'Banner__nav-button';
      }
&nbsp;
      return (
        &lt;div
          className={className}
          key={item}
          onClick={() =&gt; this.handleChangeBanner(item)}
          role="button"
        /&gt;
      );
    });
&nbsp;
&nbsp;
    return (
      &lt;div
        className="Banner"
        style={{
          backgroundImage: `url('${items[+selectedBanner].img}')`,
        }}
      &gt;
        &lt;div className="Banner__content"&gt;
          &lt;div className="Banner__logo" /&gt;
          &lt;div className="Banner__name"&gt;
            { items[+selectedBanner].name }
          &lt;/div&gt;
          &lt;div className="Banner__description"&gt;
            { items[+selectedBanner].description }
          &lt;/div&gt;
          &lt;Link
            className="Banner__button"
            to={`/recipe/${items[+selectedBanner].id}`}
          &gt;
            GET IT RECIPE
          &lt;/Link&gt;
        &lt;/div&gt;
        &lt;div className="Banner__change-menu"&gt;
          { navButtons }
        &lt;/div&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;
export default Banner;
&nbsp;
Banner.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      cookingTime: PropTypes.number.isRequired,
      description: PropTypes.string.isRequired,
      id: PropTypes.number.isRequired,
      img: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    }).isRequired,
  ).isRequired,
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Nov 11 2018 20:41:16 GMT+0300 (MSK)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
