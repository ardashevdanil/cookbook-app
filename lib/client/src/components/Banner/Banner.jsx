import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';
import './Banner.css';

class Banner extends React.Component {
  constructor(props) {
    super(props);
    this.changeBanner = this.changeBanner.bind(this);
    this.state = {
      selectedBanner: 0,
    };
    this.handleChangeBanner = this.handleChangeBanner.bind(this);
    this.timer = setInterval(this.changeBanner, 5000);
  }

  componentWillUnmount() {
    clearInterval(this.timer);
  }


  changeBanner() {
    this.setState((prevState) => {
      let nextBanner;

      if (prevState.selectedBanner === 4) {
        nextBanner = 0;
      } else {
        nextBanner = { ...prevState }.selectedBanner + 1;
      }

      return { selectedBanner: nextBanner };
    });
  }

  handleChangeBanner(id) {
    clearInterval(this.timer);
    this.setState({ selectedBanner: id });
  }

  render() {
    const { selectedBanner } = this.state;
    const { items } = this.props;

    if (!items.length) return null;

    const navButtons = [0, 1, 2, 3, 4].map((item) => {
      let className;

      if (selectedBanner === item) {
        className = 'Banner__nav-button_active Banner__nav-button';
      } else {
        className = 'Banner__nav-button';
      }

      return (
        <div
          className={className}
          key={item}
          onClick={() => this.handleChangeBanner(item)}
        />
      );
    });


    return (
      <div
        className="Banner"
        style={{
          backgroundImage: `url('${items[+selectedBanner].img}')`,
        }}
      >
        <div className="Banner__content">
          <div className="Banner__logo" />
          <div className="Banner__name">
            { items[+selectedBanner].name }
          </div>
          <div className="Banner__description">
            { items[+selectedBanner].description }
          </div>
          <Link
            className="Banner__button"
            to={`/cookbook-app/recipe/${items[+selectedBanner].id}`}
          >
            GET IT RECIPE
          </Link>
        </div>
        <div className="Banner__change-menu">
          { navButtons }
        </div>
      </div>
    );
  }
}

export default Banner;

Banner.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      cookingTime: PropTypes.number.isRequired,
      description: PropTypes.string.isRequired,
      id: PropTypes.number.isRequired,
      img: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    }).isRequired,
  ).isRequired,
};
