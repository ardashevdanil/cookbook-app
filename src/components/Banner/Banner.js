import PropTypes from 'prop-types';
import React from 'react';
import { BrowserRouter as Router, Link } from 'react-router-dom';
import './Banner.css';

import exampleImg from './exampleImg.png';

class Banner extends React.Component {
  constructor(props) {
    super(props);
    this.changeBanner = this.changeBanner.bind(this);
    this.handleChangeBanner = this.handleChangeBanner.bind(this);
    this.state = {
      selectedBanner: 0,
    };
    this.timer = setInterval(this.changeBanner, 5000);
  }

  changeBanner() {
    this.setState( (prevState) => {
      let nextBanner;

      if (prevState.selectedBanner === 4) {
        nextBanner = 0;
      } else {
        nextBanner = ++prevState.selectedBanner;
      }

      return { selectedBanner: nextBanner }
    });
  }

  componentWillUnmount() {
    clearInterval(this.timer);
  }

  handleChangeBanner(id) {
    clearInterval(this.timer);
    this.setState({ selectedBanner: id })
  }

  render() {
    if ( !this.props.items.length ) return null

    let selectedBanner = this.props.items[ this.state.selectedBanner ];
    let navButtons = [0, 1, 2, 3, 4].map( (item) => {
      let className;

      if (this.state.selectedBanner === item) {
        className = 'Banner__nav-button_active Banner__nav-button';
      } else {
        className = 'Banner__nav-button';
      }

      return (
        <div
          className={ className }
          key={ item }
          onClick={ () => this.handleChangeBanner(item) }
        />
      )
    })


    return (
      <div
        className='Banner'
        style={{
          backgroundImage: `url('${selectedBanner.img}')`,
        }}
      >
        <div className='Banner__content'>
          <div className='Banner__logo'></div>
          <div className='Banner__name'>
            { selectedBanner.name }
          </div>
          <div className='Banner__description'>
            { selectedBanner.description }
          </div>
          <Link
            className='Banner__button'
            to={`/cookbook-app/recipe/${selectedBanner.id}`}
          >
            GET IT RECIPE
          </Link>
        </div>
        <div className='Banner__change-menu'>
          { navButtons }
        </div>
      </div>
    )
  }
}

export default Banner;

Banner.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      cookingTime: PropTypes.number.isRequired,
      description: PropTypes.string.isRequired,
      id: PropTypes.number.isRequired,
      img: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    }).isRequired
  ).isRequired
}