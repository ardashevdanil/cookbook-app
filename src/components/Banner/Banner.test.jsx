import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

import Banner from './Banner';

Enzyme.configure({ adapter: new Adapter() });

describe('test Banner component', () => {
  const component = Enzyme.shallow(
    <Banner
      items={[
        {
          comments: 0,
          cookingTime: 0,
          description: '',
          difficult: '',
          id: 0,
          img: '/test',
          ingridients: ['', ''],
          name: 'test1',
          videoId: '',
          watched: 0,
        }, {
          comments: 0,
          cookingTime: 0,
          description: '',
          difficult: '',
          id: 0,
          img: '/test',
          ingridients: ['', ''],
          name: 'test1',
          videoId: '',
          watched: 0,
        },
      ]}
    />,
  );

  test('renders correctly', () => {
    expect(component.debug()).toMatchSnapshot();
  });

  test('changes state.activeBanner after click on nav button', () => {
    const states = [];

    states.push(component.state().selectedBanner);
    component.find('.Banner__nav-button').at(1).simulate('click');
    states.push(component.state().selectedBanner);

    expect(states).toEqual([0, 1]);
  });
});
