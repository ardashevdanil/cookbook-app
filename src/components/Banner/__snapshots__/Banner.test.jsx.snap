// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Banner component renders correctly 1`] = `
"<div className=\\"Banner\\" style={{...}}>
  <div className=\\"Banner__content\\">
    <div className=\\"Banner__logo\\" />
    <div className=\\"Banner__name\\">
      test1
    </div>
    <div className=\\"Banner__description\\" />
    <Link className=\\"Banner__button\\" to=\\"/cookbook-app/recipe/0\\" replace={false}>
      GET IT RECIPE
    </Link>
  </div>
  <div className=\\"Banner__change-menu\\">
    <div className=\\"Banner__nav-button_active Banner__nav-button\\" onClick={[Function: onClick]} />
    <div className=\\"Banner__nav-button\\" onClick={[Function: onClick]} />
    <div className=\\"Banner__nav-button\\" onClick={[Function: onClick]} />
    <div className=\\"Banner__nav-button\\" onClick={[Function: onClick]} />
    <div className=\\"Banner__nav-button\\" onClick={[Function: onClick]} />
  </div>
</div>"
`;
